torch/csrc/jit/batched/BatchTensor.cpp

```diff
@@ -0,0 +1,70 @@
+#include "BatchTensor.h"
+
+namespace torch { namespace jit {
+
+BatchTensor::BatchTensor(at::Tensor data, at::Tensor mask, std::vector<bool> dims){
+  if(data.dim() != mask.dim() || mask.dim() != int64_t(dims.size()) + 1){
+    throw std::runtime_error("malformed MaskedBatch with data: "
+      + std::string(data.toString()) + " mask: " + mask.toString());
+  }
+  this->data = data;
+  this->mask = mask;
+  this->dims = dims;
```

Comment:
Since you take `dims` by value, move out of the copy. `this->dims = std::move(dims);`